### ISSUE TITLE
Zigbee2mqtt ls: cannot access '/dev/serial/by-id': No such file or directory

### ISSUE THREAD

**Issue by MRobi1** [tteck/Proxmox#30]
Trying to use your zigbee2mqtt script to setup an LXC on proxmox.

Ran steps 1 and 2 to create the LXC and allow USB passthrough.
Reboot LXC.
Installed usbutils and ran the lsusb command and I do see the USB device there.
`Bus 002 Device 071: ID 1cf1:0030 Dresden Elektronik ZigBee gateway [ConBee II]`

However when I run
`ls -l /dev/serial/by-id`

It returns
`ls: cannot access '/dev/serial/by-id': No such file or directory`

**Comment by tteck** [comment]
Can you post your `.conf` for that LXC?

**Comment by MRobi1** [comment]
Here you go

```
arch: amd64
cores: 2
features: nesting=1
hostname: zigbee2mqtt
memory: 1024
net0: name=eth0,bridge=vmbr0,hwaddr=B6:6F:9C:46:9E:21,ip=dhcp,type=veth
onboot: 1
ostype: debian
rootfs: local-lvm:vm-110-disk-0,size=4G
swap: 512
lxc.cgroup2.devices.allow: a
lxc.cap.drop: 
lxc.autodev: 1
lxc.hook.autodev: bash -c 'for char_dev in $(find /sys/dev/char -regextype sed  -regex ".*/188:.*" -o -regex ".>
```

**Comment by tteck** [comment]
Looks correct.
What's the output of `ls -l /dev/ttyUSB0`?

**Comment by MRobi1** [comment]
Same
`ls: cannot access '/dev/ttyUSB0': No such file or directory`

**Comment by tteck** [comment]
Replace the last two lines in the `.conf` with
```yaml
lxc.cgroup2.devices.allow: c 188:* rwm
lxc.cgroup2.devices.allow: c 189:* rwm
lxc.mount.entry: /dev/serial/by-id               dev/serial/by-id         none bind,optional,create=dir
lxc.mount.entry: /dev/ttyUSB0                    dev/ttyUSB0             none bind,optional,create=file
lxc.mount.entry: /dev/ttyACM0                    dev/ttyACM0             none bind,optional,create=file
```
Reboot the LXC and share the output of `ls -l /dev/serial/by-id`

**Comment by MRobi1** [comment]
Getting somewhere...
```
root@zigbee2mqtt:~# ls -l /dev/serial/by-id
total 0
lrwxrwxrwx 1 root root 13 Feb  9 20:02 usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2212845-if00 -> ../../ttyACM0
```

Zigbee2mqtt fails to start
```
Zigbee2MQTT:info  2022-02-09 20:09:30: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2022-02-09.20-09-29' filename: log.txt
Zigbee2MQTT:info  2022-02-09 20:09:30: Starting Zigbee2MQTT version 1.23.0 (commit #6ea61bae)
Zigbee2MQTT:info  2022-02-09 20:09:30: Starting zigbee-herdsman (0.14.10)
Zigbee2MQTT:error 2022-02-09 20:09:30: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2022-02-09 20:09:30: Failed to start zigbee
Zigbee2MQTT:error 2022-02-09 20:09:30: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2022-02-09 20:09:30: Exiting...
Zigbee2MQTT:error 2022-02-09 20:09:30: Error: Error while opening serialport 'Error: Error Resource temporarily unavailable Cannot lock port'
    at SerialPort.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/znp/znp.ts:146:28)
    at SerialPort._error (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/node_modules/@serialport/stream/lib/index.js:198:14)
    at /opt/zigbee2mqtt/node_modules/zigbee-herdsman/node_modules/@serialport/stream/lib/index.js:242:12
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zigbee2mqtt@1.23.0 start: `node index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zigbee2mqtt@1.23.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-02-10T00_09_30_669Z-debug.log
```

In configuration.yaml I tried both port: /dev/ttyACM0 and port: /dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2212845-if00
Same message with both.

**Comment by tteck** [comment]
ConBeeII needs 
```yaml
serial:
  port: /dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2212845-if00
  adapter: deconz
```

**Comment by MRobi1** [comment]
Added the adapter: deconz to the configuration.yaml and same error. Cannot lock port.

**Comment by tteck** [comment]
Are you using the ConBee II with ZHA while setting up Z2M? Only 1 zigbee network per coordinator.

**Comment by MRobi1** [comment]
Nope. I've got a CC2652 on another instance of z2m running on my HASS box. 
Looking to set this up as a test network and had the Conbee laying around. Way back I used to have it connected to deconz though.

**Comment by tteck** [comment]
Just guessing here (never used a ConBee), but I'd say the firmware needs updating.

**Comment by MRobi1** [comment]
Got it :)
A little digging through the z2m issue reports and I found this gem.

"Faced the same issue. I tried to run npm start from the project folder. But got an error. This happens due to already executed zigbee2mqtt.service via systemctl."

Sure enough, navigating to http://192.168.XX.XXX:9442 and it was already running!!

Thanks for all the help with this one

**Comment by tteck** [comment]
BTW, I just ahead and added
```yaml
lxc.cgroup2.devices.allow: c 188:* rwm
lxc.cgroup2.devices.allow: c 189:* rwm
lxc.mount.entry: /dev/serial/by-id  dev/serial/by-id  none bind,optional,create=dir
lxc.mount.entry: /dev/ttyUSB0       dev/ttyUSB0       none bind,optional,create=file
lxc.mount.entry: /dev/ttyACM0       dev/ttyACM0       none bind,optional,create=file
```
to the `.conf` file during the installation. Should make it a more pleasant setup process.

**Comment by alamoudimoh** [comment]
> BTW, I just went ahead and added
> 
> ```yaml
> lxc.cgroup2.devices.allow: c 188:* rwm
> lxc.cgroup2.devices.allow: c 189:* rwm
> lxc.mount.entry: /dev/serial/by-id  dev/serial/by-id  none bind,optional,create=dir
> lxc.mount.entry: /dev/ttyUSB0       dev/ttyUSB0       none bind,optional,create=file
> lxc.mount.entry: /dev/ttyACM0       dev/ttyACM0       none bind,optional,create=file
> ```
> 
> to the `.conf` file during the installation. Should make it a more pleasant setup process.

i have the same setup using sonoff 3 and the same error 

ls -l /dev/serial/by-id
shows total 0

lxc.cgroup2.devices.allow: c 188:* rwm
lxc.cgroup2.devices.allow: c 189:* rwm
lxc.mount.entry: /dev/serial/by-id  dev/serial/by-id  none bind,optional,create=dir
lxc.mount.entry: /dev/ttyUSB0       dev/ttyUSB0       none bind,optional,create=file
lxc.mount.entry: /dev/ttyUSB1       dev/ttyUSB1       none bind,optional,create=file
lxc.mount.entry: /dev/ttyACM0       dev/ttyACM0       none bind,optional,create=file
lxc.mount.entry: /dev/ttyACM1       dev/ttyACM1       none bind,optional,create=file

**Comment by tteck** [comment]
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1035094

https://github.com/tteck/Proxmox/discussions/1388#discussioncomment-5796713

**Comment by alamoudimoh** [comment]
> ls -l /dev/ttyUSB0

i managed it,
however when i start z2m
i get the following
Zigbee2MQTT:info  2023-05-08 22:23:51: Connecting to MQTT server at mqtt://192.168.1.221:1883
Zigbee2MQTT:error 2023-05-08 22:23:51: MQTT error: connect ECONNREFUSED 192.168.1.221:1883
Zigbee2MQTT:error 2023-05-08 22:23:51: MQTT failed to connect, exiting...
Zigbee2MQTT:info  2023-05-08 22:23:51: Stopping zigbee-herdsman...
Zigbee2MQTT:error 2023-05-08 22:23:52: MQTT error: connect ECONNREFUSED 192.168.1.221:1883
Zigbee2MQTT:error 2023-05-08 22:23:53: MQTT error: connect ECONNREFUSED 192.168.1.221:1883
Zigbee2MQTT:error 2023-05-08 22:23:54: MQTT error: connect ECONNREFUSED 192.168.1.221:1883
Zigbee2MQTT:error 2023-05-08 22:23:55: MQTT error: connect ECONNREFUSED 192.168.1.221:1883
Zigbee2MQTT:error 2023-05-08 22:23:56: MQTT error: connect ECONNREFUSED 192.168.1.221:1883
Zigbee2MQTT:error 2023-05-08 22:23:57: MQTT error: connect ECONNREFUSED 192.168.1.221:1883
Zigbee2MQTT:info  2023-05-08 22:23:57: Stopped zigbee-herdsman
root@zigbee2mqtt:/opt/zigbee2mqtt#

**Comment by tteck** [comment]
Is it possible that the credentials for connecting to the MQTT server are not configured properly, which is causing the connection to be refused? It's a good idea to check the MQTT server configuration and make sure that the credentials (e.g., username and password) are set up correctly and match the credentials being used by Zigbee2MQTT to connect to the server.

**Comment by TurboCoder13** [comment]
Sorry for reviving an old thread! 😅 I just want to share the solution that worked for me.

A while ago, I found a solution where you had to edit one of the files in the `/lib/udev/rules.d` but I couldn't find this solution again.

With a more recent search, the [solution](https://www.reddit.com/r/pop_os/comments/uf54bi/comment/i6rifto/?utm_source=share&utm_medium=web2x&context=3) is (IMO) simpler.

> _you can disable it:_
> 
> ```
> systemctl stop brltty-udev.service
> sudo systemctl mask brltty-udev.service
> systemctl stop brltty.service
> systemctl disable brltty.service
> ```

Wrt this:

> _I don't recall if you have to reboot afterwards._

You don't need to (at least I didn't need to 🙂).

# tl;dr: 

> _`brltty` provides access to the console screen (when in text mode) for a blind person using a refreshable braille display_

So unless you absolutely need the braille support, I think it should be fine to disable it.

# Edit:

I am running Z2M with docker-compose on Ubuntu LTS.

**Comment by tteck** [comment]
There are no such services on Proxmox
```
systemctl list-unit-files
```

**Comment by marcj3n** [comment]
Hi,

I have a similar issue, hope someone can help me out.  It is the first time I'm using proxmox but I tried the different solutions posted here and in other channels but cannot seem to get this to work.

When I run
`ls -l /dev/serial/by-id`

It returns
`ls: cannot access '/dev/serial/by-id': No such file or directory`

lsusb -v returns
```
Bus 001 Device 003: ID 10c4:ea60 Silicon Labs CP210x UART Bridge
... 
iProduct                2 Sonoff Zigbee 3.0 USB Dongle Plus
... 
```
My .conf for zigbee2mqtt:
```
arch: amd64
cores: 2
features: nesting=1
hostname: zigbee2mqtt
memory: 1024
net0: name=eth0,bridge=vmbr0,hwaddr=BC:24:11:6B:EB:F7,ip=dhcp,type=veth
onboot: 1
ostype: debian
rootfs: local-lvm:vm-101-disk-0,size=4G
swap: 512
tags: proxmox-helper-scripts
lxc.cgroup2.devices.allow: a
lxc.cap.drop:
lxc.cgroup2.devices.allow: c 188:* rwm
lxc.cgroup2.devices.allow: c 189:* rwm
lxc.mount.entry: /dev/serial/by-id  dev/serial/by-id  none bind,optional,create=dir
lxc.mount.entry: /dev/ttyUSB0       dev/ttyUSB0       none bind,optional,create=file
lxc.mount.entry: /dev/ttyUSB1       dev/ttyUSB1       none bind,optional,create=file
lxc.mount.entry: /dev/ttyACM0       dev/ttyACM0       none bind,optional,create=file
lxc.mount.entry: /dev/ttyACM1       dev/ttyACM1       none bind,optional,create=file
```
I executed this successfully but it did not change my outcome above: https://github.com/tteck/Proxmox/discussions/1388#discussioncomment-5796713 

Thanks!

Marc
